packages/core/src/Canvas.ts

```diff
@@ -429,4 +429,23 @@ export class Canvas<T extends Contract = Contract> extends TypedEventEmitter<Can
 	): AsyncIterable<[id: string, signature: Signature, message: Message<Action | Session>]> {
 		yield* this.messageLog.iterate(lowerBound, upperBound, options)
 	}
+
+	/**
+	 * Get an existing session
+	 */
+	public async getSession(query: { address: string; publicKey: string; timestamp?: number }): Promise<string | null> {
+		const sessions = await this.db.query<{ message_id: string }>("$sessions", {
+			where: {
+				public_key: query.publicKey,
+				address: query.address,
+				expiration: { gte: query.timestamp ?? 0 },
+			},
+		})
+
+		if (sessions.length === 0) {
+			return null
+		} else {
+			return sessions[0].message_id
+		}
+	}
```

Comment:
As a note, Joel is probably going to look at this, and try to figure out a way to simplify getting existing sessions from the internal store.